### DISCUSSION TITLE
.NET MAUI - Android API Level 31 Required?

### DISCUSSION THREAD

**Discussion by tomshane** [reactiveui/ReactiveUI#3315] [General]
When I'm trying to compile the project, I'm getting warning:

`'ReactiveCommand.CreateFromTask(Func<Task>, IObservable<bool>?, IScheduler?)' is only supported on: 'Android' 31.0 and later.`

Is this really that using Reactive UI with .NET MAUI needs devices with API Level 31 (Android 12+)? This is very restrictive, only ~15% of people would be able to use the application based on Reactive UI.

**Comment by glennawatson** [comment]
https://developer.android.com/google/play/requirements/target-sdk#:~:text=When%20you%20upload%20an%20APK,API%20level%2028%20or%20higher.

Requirement of google not maui

> Starting in August 2022, new apps must target API level 31 (Android 12) or above and adjust for behavioral changes. Wear OS apps must target API level 28 or higher.

**Reply by glennawatson** [reply]
Also note targetting API Level 31 doesn't mean you can't set a minimum API level to much lower, just means you have to build against API Level 31

**Reply by rgomez90** [reply]
@glennawatson I still not really understand what we are supposed to use a target framework and min. target framework.

* If I use as target framework API Level 32 or API Level 33 (latest API Level) VS tells me everything is not available on that level
![image](https://user-images.githubusercontent.com/12009909/185756439-65fab4c2-11bd-4c47-9f3e-d60d9ec4c6b4.png)

* If I use target framework API Level 31 and a lower API Level (for ex. 28) as min. target framework, the warning for Android goes away

Now for iOS I get this warning 
![image](https://user-images.githubusercontent.com/12009909/185756570-6d46db45-3b3e-4014-a958-1726b00037a5.png)
If I set target framework 15.4 and a lower min. target framework (for ex. 14.2) the warning does NOT go away....
So what should I do to get rid of the warnings in both platforms or am I doing something wrong / misunderstanding something?

**Comment by tomshane** [comment]
Oh, I completely misunderstood the warning then, my bad. Thanks for the explanation!

**Reply by glennawatson** [reply]
Yeah, if you build against Android31 btw, it just means you use that API surface, it's backwards compatible with older phone devices.

So generally best if you can to target the most recent production android API, since Google generally force you to sooner rather than later as well.

**Comment by ChrisPulman** [comment]
You can use the following project properties to configure your versions for each MAUI platform.
SupportedOSPlatformVersion
TargetPlatformMinVersion

**Reply by tomshane** [reply]
Thanks. Setting TargetPlatformMinVersion made the warning go away, however I had to set it directly in csproj file, it doesn't seem to be available from the VS UI.

**Comment by ChrisPulman** [comment]
Yes correct, not all the MAUI tools and settings are in Visual Studio yet. Using the latest preview version of Visual Studio is probably your best option to obtain MAUI tools and settings earlier.
